### ISSUE TITLE
Feature Request: Support BigQuery clustering

### ISSUE THREAD

**Issue by migibert** [terraform-providers/terraform-provider-google#1852]
Hi,

It would be great if we could create BigQuery tables with setting clustering columns.
Documentation: https://cloud.google.com/bigquery/docs/creating-clustered-tables

### Affected Resource(s)
- google_bigquery_table

**Comment by migibert** [comment]
Hi,

I wanted to work on this feature, but it looks like the bigquery library in vendors is not the latest one.

I tried to upgrade it using `govendor update google.golang.org/api/bigquery/v2` and I can observe changes on the following files:
- vendor/google.golang.org/api/bigquery/v2/bigquery-api.json
- vendor/google.golang.org/api/bigquery/v2/bigquery-gen.go
- vendor/vendor.json

So it looked good to me but now the provider does not build because of this error:
```
make build
==> Checking that code complies with gofmt requirements...
go install
# github.com/terraform-providers/terraform-provider-google/vendor/google.golang.org/api/bigquery/v2
vendor/google.golang.org/api/bigquery/v2/bigquery-gen.go:5466:25: assignment mismatch: 3 variables but 2 values
vendor/google.golang.org/api/bigquery/v2/bigquery-gen.go:5471:2: undefined: gensupport.SetGetBody
GNUmakefile:9: recipe for target 'build' failed
make: *** [build] Error 2
```
Can you give me some directions about how to fix this ? Maybe I did not upgrade the library correctly ?

**Comment by ndmckinley** [comment]
Govendor is a little odd.  You probably wanted `govendor fetch` instead of `govendor update`.  Does it work if you do that?

**Comment by migibert** [comment]
Hey, thanks for your help. Sadly, I tried with fetch command and met the same problem...

**Comment by ndmckinley** [comment]
Ah... In the docs for that package:
>This package is DEPRECATED. Use package cloud.google.com/go/bigquery instead.

I'm not sure if that's really going to make this easier, though.  It seems like the API is completely different, and switching would require a complete rewrite of the existing code.  I think it would be tough to take that on.

**Comment by migibert** [comment]
Thank you ! I saw the deprecation warning but hoped I could still use this library... :'(

**Comment by ndmckinley** [comment]
I definitely did not find a solution - please, feel free.  :)

**Comment by pashashiz** [comment]
Hello, any updates on this?

**Comment by ndmckinley** [comment]
I believe this is unlocked now that the BigQuery team has generated a REST client, but I don't believe that anyone is currently working on this - we'd definitely accept a community contribution, and we'll put it on our radar for prioritization as well.

**Comment by pashashiz** [comment]
What about the the @Cidan's pull request? Any idea why that was closed.

**Comment by ndmckinley** [comment]
It became obsolete when we moved this repo to be fully generated in Magic Modules - it could be reopened against that repo, I believe.

**Comment by Tho77** [comment]
Thought I'd pile on support for this enhancement so I don't have to create these tables manually outside of TF.

**Comment by nathadfield** [comment]
Adding some further weight behind this.  Not supporting clustering puts TF at a disadvantage.

**Comment by ashuclh** [comment]
Any updates on this?

**Comment by kvas-damian** [comment]
Another PR trying to address this issue: https://github.com/terraform-providers/terraform-provider-google/pull/4223

**Comment by kvas-damian** [comment]
I think the issue is solved by https://github.com/terraform-providers/terraform-provider-google/pull/4223

**Comment by megan07** [comment]
Thanks @migibert for opening this issue. It's been closed with #4223 and has been released in 2.14.0.
